Dockerfile

```
@@ -54,49 +52,38 @@ ARG DALI_TIMESTAMP
 # Optional build arguments
 
 ARG CMAKE_BUILD_TYPE=Release
+ENV CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
 ARG BUILD_TEST=ON
+ENV BUILD_TEST=${BUILD_TEST}
 ARG BUILD_BENCHMARK=ON
+ENV BUILD_BENCHMARK=${BUILD_BENCHMARK}
 ARG BUILD_NVTX=OFF
+ENV BUILD_NVTX=${BUILD_NVTX}
 ARG BUILD_PYTHON=ON
+ENV BUILD_PYTHON=${BUILD_PYTHON}
 ARG BUILD_LMDB=ON
+ENV BUILD_LMDB=${BUILD_LMDB}
 ARG BUILD_TENSORFLOW=ON
+ENV BUILD_TENSORFLOW=${BUILD_TENSORFLOW}
 ARG BUILD_JPEG_TURBO=ON
+ENV BUILD_JPEG_TURBO=${BUILD_JPEG_TURBO}
 ARG BUILD_NVJPEG=ON
+ENV BUILD_NVJPEG=${BUILD_NVJPEG}
 ARG BUILD_NVOF=ON
+ENV BUILD_NVOF=${BUILD_NVOF}
 ARG BUILD_NVDEC=ON
+ENV BUILD_NVDEC=${BUILD_NVDEC}
 ARG BUILD_NVML=ON
-
-RUN LD_LIBRARY_PATH="${PWD}:${LD_LIBRARY_PATH}" && \
-    cmake ../ -DCMAKE_INSTALL_PREFIX=. -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
-          -DBUILD_TEST=${BUILD_TEST} -DBUILD_BENCHMARK=${BUILD_BENCHMARK} \
-          -DBUILD_NVTX=${BUILD_NVTX} -DBUILD_PYTHON=${BUILD_PYTHON} \
-          -DBUILD_LMDB=${BUILD_LMDB} -DBUILD_TENSORFLOW=${BUILD_TENSORFLOW}  \
-          -DBUILD_JPEG_TURBO=${BUILD_JPEG_TURBO} -DBUILD_NVJPEG=${BUILD_NVJPEG} \
-          -DBUILD_NVOF=${BUILD_NVOF} -DBUILD_NVDEC=${BUILD_NVDEC} \
-          -DBUILD_NVML=${BUILD_NVML} \
-          -DWERROR=ON \
-          -DDALI_BUILD_FLAVOR=${NVIDIA_DALI_BUILD_FLAVOR} \
-          -DTIMESTAMP=${DALI_TIMESTAMP} -DGIT_SHA=${GIT_SHA} && \
-    make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+ENV BUILD_NVML=${BUILD_NVML}
 
 ARG NVIDIA_BUILD_ID
 ENV NVIDIA_BUILD_ID ${NVIDIA_BUILD_ID:-0}
+RUN mkdir /wheelhouse && chmod 0777 /wheelhouse
+
+FROM builder
+COPY . .
+
+ARG DALI_BUILD_DIR=build-docker-release
+WORKDIR /opt/dali/${DALI_BUILD_DIR}
 
-RUN if [ "${BUILD_PYTHON}" = "ON" ]; then \
-        pip wheel -v dali/python \
-            --build-option --python-tag=$(basename /opt/python/cp${PYV}-*) \
-            --build-option --plat-name=manylinux1_x86_64 \
-            --build-option --build-number=${NVIDIA_BUILD_ID} && \
-        ../dali/python/bundle-wheel.sh nvidia_dali[_-]*.whl && \
-        UNZIP_PATH="$(mktemp -d)" && \
-        unzip /wheelhouse/nvidia_dali*.whl -d $UNZIP_PATH && \
-        python ../tools/test_bundled_libs.py $(find $UNZIP_PATH -iname *.so* | tr '\n' ' ') && \
-        rm -rf $UNZIP_PATH ;\
-    fi
-
-RUN if [ "${BUILD_PYTHON}" = "ON" ]; then \
-        pushd dali/python/tf_plugin/ && \
-        python setup.py sdist && \
-        mv dist/nvidia-dali-tf-plugin*.tar.gz /wheelhouse/ && \
-        popd ;\
-    fi
+RUN /opt/dali/build.sh
```

As we are changing this, can we make it 
```suggestion
CMD /opt/dali/build.sh
```
This allows you to build an image and run it, and the CMD will be executed by default, but you can overwrite it, and have a interacitve run:
run -it dali_builder /bin/bash

---

We can but we don't want to to build such docker just to run it later as it is time consuming without any additional value. Also CMD is run-time command, while RUN will produce layer with binaries we can copy out.

---

Sure, I thought it was more suited for the build that uses mounted volume instead of copied sources.